src/scripts/booking.js

```diff
@@ -0,0 +1,191 @@
+
+// Book a room (Second Step)
+function bookStepTwo() {
+  const bookingPartOneDiv = document.querySelector('#bookingPartOne');
+  body.removeChild(bookingPartOneDiv);
+
+  const secondTemplate = `
+  <div class="booking" id="bookingPartTwo">
+    <div class="second-modal">
+        <h2>Book room "${challengeTitle}" (Step 2)</h2>
+
+        <label for="name">Name</label>
+        <input type="text" name="name" id="name">
+
+        <label for="email">E-mail</label>
+        <input type="email" name="email" id="email">
+
+        <label for="time">What time?</label>
+        <select name="time" id="time"></select>
+
+        <label for="participants">How many participants?</label>
+        <select name="participants" id="participants">
+            <option value="2">2 participants</option>
+            <option value="3">3 participants</option>
+            <option value="4">4 participants</option>
+            <option value="5">5 participants</option>
+            <option value="6">6 participants</option>
+        </select>
+
+        <div class="btn-parent">
+        <button id="submitBtnTwo" class="button primary">Submit booking</button>
+    </div>
+    </div>
+  </div>
+  `;
+ 
+  body.insertAdjacentHTML("afterbegin", secondTemplate);
+
+  let dateTemplate;
+  times.forEach((time)=> {
+    dateTemplate += `<option value="${time}">${time}</option>`
+  });
+
+  const time = document.querySelector("#time");
+  time.insertAdjacentHTML("afterbegin", dateTemplate);
+
+  // Submit
+  const submitBtnTwo = document.querySelector("#submitBtnTwo");
+  submitBtnTwo.addEventListener('click', () => {
+    const name = document.querySelector('#name').value;
+    const email = document.querySelector('#email').value;
+
+    // Time selected
+    const timeDiv = document.querySelector('#time');
+    const timeSelected = timeDiv.options[timeDiv.selectedIndex].text;
+
+    // Participants
+    let particiDiv = document.querySelector('#participants');
+    let participantSelected = particiDiv.options[particiDiv.selectedIndex].value;
+    participantSelected = parseInt(participantSelected);
+
+    submitStepTwo(challengeNumber, name,  email, date, timeSelected, participantSelected);
+  });
+
+  addCancelOption();
+}
+
+// Submit Step 2
+async function submitStepTwo(challenge, name, email, date, time, participants) {
+  if(!name || !email) {
+    alert('Fill all the fields before submitting!');
+    return;
+  }
+  const res = await fetch(`${api}/booking/reservations`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      challenge,
+      name,
+      email,
+      date,
+      time,
+      participants,
+    }),
+  });
+
+  const data = await res.json();
+  finalRes = data;
+  bookStepThree();
+}
+
+// Book a room (Third Step)
+function bookStepThree() {
+  const bookingPartTwoDiv = document.querySelector('#bookingPartTwo');
+  body.removeChild(bookingPartTwoDiv);
+
+  const thirdTemplate = `
```

Comment:
I like the way your HTML-template looks and is written. Nice way to do it.

Reply:
Thanks man